internal/service/x509authenticationdatabaseuser/resource_x509_authentication_database_user.go

```diff
@@ -176,21 +176,8 @@ func resourceMongoDBAtlasX509AuthDBUserRead(ctx context.Context, d *schema.Resou
 }
 
 func resourceMongoDBAtlasX509AuthDBUserDelete(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
-	conn := meta.(*config.MongoDBClient).Atlas
-
-	ids := conversion.DecodeStateID(d.Id())
-	currentCertificate := ids["current_certificate"]
-	projectID := ids["project_id"]
-
-	if currentCertificate == "" {
-		_, err := conn.X509AuthDBUsers.DisableCustomerX509(ctx, projectID)
-		if err != nil {
-			return diag.FromErr(fmt.Errorf(errorCustomerX509AuthDBUsersDelete, projectID, err))
-		}
-	}
-
+	// We don't do anything because X.509 certificates can not be deleted or disassociated from a user
```

Comment:
Is this a breaking change? Do we want to hold it on after we release 1.14.0?

Reply:
I guess it's not a breaking change because it'd return error now...

Reply:
no, on the contrary, the client wants this as soon as possible.   The behavior will change, but for the good, because current behavior is unexpected

Reply:
I understand we are limited in what we can for this scenario, but just want to make sure there are no significant side effects.
From what I understand if a user now deletes the `mongodbatlas_x509_authentication_database_user` the "Self-Managed X.509 Authentication" would still be enabled, and the users would have to disable manually if that is his intention.

Reply:
yes.  but the normal use case is to have multiple users, so when you disable one you don't want to disable others.
As an example in Atlas UI when you delete the database user, the X.509 keeps enabled.  Here we're doing the same behavior as in the UI.

Reply:
there is useful info in the comments in: https://jira.mongodb.org/browse/HELP-53363

Reply:
doc updated